tooling/karaf-maven-plugin/src/main/java/org/apache/karaf/tooling/features/GenerateDescriptorMojo.java

```diff
@@ -686,6 +686,9 @@ private Manifest getManifest(File file) throws IOException {
                 }
                 return m;
             }
+        } catch (IOException e) {
+            getLog().warn("Error while reading artifact", e);
+            return null;
```

Comment:
I would prefer to keep `throws IOException` and add log. Something like;

```
private Manifest getManifest(File file) throws IOException {
  final InputStream is;
  try {
    is = Files.newInputStream(file.toPath());
    ...
  } catch (IOException e) {
    getLog().warn("Error while reading artifact", e);
    throw e;
 }
```

Reply:
Unfortunately that doesn't solve the problem I'm having.  That same exception in v4.1.2 would have been turned into a warning at https://github.com/apache/karaf/pull/708/files#diff-0a5a9e3c3d44c1b639a942a084b31627L676 and not re-thrown.  I'm really trying to get back to that behaviour.

Reply:
I don't understand: my proposal is to both log AND rethrow. So it should work as you expect (as you expect a warn).

Reply:
Yeah, whether the warning happens or not, I need the "Is a directory" IOException to not show up outside of this method.   Previously the exception would be thrown on opening the FileInputStream, so the exception would get caught at line 674 and the method would return null.  Now (after converting to NIO) the exception is thrown when the stream is read from, so it is thrown on line 682 and is never caught.  That's why I'm adding the catch block where I am and returning null.
I hope that's more clear?